Core/test/qa-functional/src/org/sleuthkit/autopsy/ingest/EmbeddedFileTest.java

```diff
@@ -92,31 +94,47 @@ public void tearDown() {
         CaseUtils.closeCurrentCase(testSucceeded);
     }
     
-    public void testEncryption() {
+    public void testEncryptionAndZipBomb() {
         try {
-            List<AbstractFile> results = openCase.getSleuthkitCase().findAllFilesWhere("name LIKE '%%'");            
-            String protectedName1 = "password_protected.zip";
-            String protectedName2 = "level1_protected.zip";
-            String protectedName3 =  "42.zip";
-            assertEquals(2207, results.size());
+            List<AbstractFile> results = openCase.getSleuthkitCase().findAllFilesWhere("name LIKE '%%'");
+            final String zipBombSetName = "Possible Zip Bomb";
+            final String protectedName1 = "password_protected.zip";
+            final String protectedName2 = "level1_protected.zip";
+            final String protectedName3 =  "42.zip"; 
+            final String depthZipBomb = "DepthTriggerZipBomb.zip";
+            final String ratioZipBomb = "RatioTriggerZipBomb.zip";
+            int zipBombs = 0;
+            assertEquals("The number of files in the test image has changed", 2221, results.size());
             int passwdProtectedZips = 0;
             for (AbstractFile file : results) {
                 //.zip file has artifact TSK_ENCRYPTION_DETECTED
                 if (file.getName().equalsIgnoreCase(protectedName1) || file.getName().equalsIgnoreCase(protectedName2) || file.getName().equalsIgnoreCase(protectedName3)){
                     ArrayList<BlackboardArtifact> artifacts = file.getAllArtifacts();
-                    assertEquals(1, artifacts.size());
+                    assertEquals("Password protected zip file " + file.getName() + " has incorrect number of artifacts", 1, artifacts.size());
                     for (BlackboardArtifact artifact : artifacts) {
-                        assertEquals(artifact.getArtifactTypeID(), BlackboardArtifact.ARTIFACT_TYPE.TSK_ENCRYPTION_DETECTED.getTypeID());
+                        assertEquals("Artifact for password protected zip file " + file.getName() + " has incorrect type ID", artifact.getArtifactTypeID(), BlackboardArtifact.ARTIFACT_TYPE.TSK_ENCRYPTION_DETECTED.getTypeID());
                         passwdProtectedZips++;
                     }
+                } else if (file.getName().equalsIgnoreCase(depthZipBomb) || file.getName().equalsIgnoreCase(ratioZipBomb)){
+                    ArrayList<BlackboardArtifact> artifacts = file.getAllArtifacts();
+                    assertEquals("Zip bomb " + file.getName() + " has incorrect number of artifacts", 1, artifacts.size());
+                    for (BlackboardArtifact artifact : artifacts) {
```

Comment:
I always think a good best practice to follow is to put "List" at the name of a List object. So here, you would have 'artifactsList'. It makes it stand out a little better so the person reading the code doesn't misread it.

Reply:
I've personally always found it clear enough to interpret it as plural -> some kind of collection.

Reply:
@wschaeferB and @dgrove727, I agree with @APriestman . I find reading "artifacts" to be easier than reading "artifactsList." To me its a matter of level of abstraction. Reading the code to understand it, all I need to know is there is some collection of artifacts. If I care about the type of the collection, I look at the variable declaration, although that is rarely necessary. I try to make variable names be like what we would use in conversation or pseudo-code, to make code closer to speech...